File: transfer/Int2NumeralExample.hs
module Int2NumeralExample where

import Int2Numeral
import PGF

-- to test: echo "you have 500 messages" | runghc Int2NumeralExample.hs

main = do
  pgf <- readPGF "Int2NumeralExample.pgf"
  let eng:_ = languages pgf 
  interact (unlines . map (linearize pgf eng) . map transfer . parse pgf eng (startCat pgf))

transfer = int2numeralInTree [
  (mkCId "IntNumber", mkCId "NumeralNumber"),
  (mkCId "IntRank", mkCId "NumeralRank")
  ]

File: transfer/int2numeral_example.py
from int2numeral import int2numeral_in_tree
import pgf

# to test: echo "you have 500 messages" | python3 int2numeral_example.py

int_numeral_dict = {"IntNumber": "NumeralNumber", "IntRank": "NumeralRank"}

def main():
    gr = pgf.readPGF("Int2NumeralExample.pgf")
    eng = list(gr.languages.values())[0]
    _,tree = eng.parse(input()).__next__()
    tree = int2numeral_in_tree(int_numeral_dict, tree)
    print(eng.linearize(tree))


main()

